internal/server/server.go

```diff
@@ -0,0 +1,280 @@
+
+	idenfyClient := idenfy.New(&s.config.Idenfy, s.logger)
+
+	substrateClient, err := substrate.New(&s.config.TFChain, s.logger)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize substrate client: %w", err)
+	}
+
+	return services.NewKYCService(
+		repos.verification,
+		repos.token,
+		idenfyClient,
+		substrateClient,
+		s.config,
+		s.logger,
+	), nil
+}
+
+func (s *Server) setupRoutes(kycService services.KYCService, mongoCl *mongo.Client) error {
+	s.logger.Debug("Setting up routes", nil)
+
+	handler := handlers.NewHandler(kycService, s.config, s.logger)
+
+	// API routes
+	v1 := s.app.Group("/api/v1")
+	v1.Post("/token", middleware.AuthMiddleware(s.config.Challenge), handler.GetorCreateVerificationToken())
+	v1.Get("/data", middleware.AuthMiddleware(s.config.Challenge), handler.GetVerificationData())
+	v1.Get("/status", handler.GetVerificationStatus())
+	v1.Get("/health", handler.HealthCheck(mongoCl))
+	v1.Get("/configs", handler.GetServiceConfigs())
+	v1.Get("/version", handler.GetServiceVersion())
+
+	// Webhook routes
+	webhooks := s.app.Group("/webhooks/idenfy")
+	webhooks.Post("/verification-update", handler.ProcessVerificationResult())
+	webhooks.Post("/id-expiration", handler.ProcessDocExpirationNotification())
+
+	// Documentation
+	s.app.Get("/docs/*", swagger.HandlerDefault)
+
+	return nil
+}
+
+func extractIPFromRequest(c *fiber.Ctx) string {
+	// Check for X-Forwarded-For header
+	if ip := c.Get("X-Forwarded-For"); ip != "" {
+		ips := strings.Split(ip, ",")
+		if len(ips) > 0 {
+
+			for _, ip := range ips {
+				// return the first non-private ip in the list
+				if net.ParseIP(strings.TrimSpace(ip)) != nil && !net.ParseIP(strings.TrimSpace(ip)).IsPrivate() {
+					return strings.TrimSpace(ip)
+				}
+			}
+		}
+	}
+	// Check for X-Real-IP header if not a private IP
+	if ip := c.Get("X-Real-IP"); ip != "" {
+		if net.ParseIP(strings.TrimSpace(ip)) != nil && !net.ParseIP(strings.TrimSpace(ip)).IsPrivate() {
+			return strings.TrimSpace(ip)
+		}
+	}
+	// Fall back to RemoteIP() if no proxy headers are present
+	ip := c.IP()
+	if parsedIP := net.ParseIP(ip); parsedIP != nil {
+		if !parsedIP.IsPrivate() {
+			return ip
+		}
+	}
+	// If we still have a private IP, return a default value that will be skipped by the limiter
+	return "127.0.0.1"
+}
+
+func (s *Server) Start() {
+	go func() {
+		sigChan := make(chan os.Signal, 1)
+		signal.Notify(sigChan, os.Interrupt, syscall.SIGTERM)
+		<-sigChan
+		// Graceful shutdown
+		s.logger.Info("Shutting down server...", nil)
+		ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+		defer cancel()
+		if err := s.app.ShutdownWithContext(ctx); err != nil {
+			s.logger.Error("Server forced to shutdown:", logger.Fields{"error": err})
+		}
+	}()
+
+	// Start server
+	if err := s.app.Listen(":" + s.config.Server.Port); err != nil && err != http.ErrServerClosed {
+		s.logger.Fatal("Server startup failed", logger.Fields{"error": err})
```

Comment:
don't terminate the application from here, proagate the error and the caller should decide (should be in the main function). Keep single exit point for your application